lib/api/bucketPutVersioning.js

```diff
@@ -17,6 +18,36 @@ import services from '../services';
     x-amz-mfa: [SerialNumber] [TokenCode]
  */
 
+function _parseXML(request, log, cb) {
+    if (request.post === '') {
```

Comment:
`if (!request.post)` check should be sufficient here. Also, need to confirm if `MalformedXML` is the correct error.